File: RNShimmer/RNShimmer.m
//
//  RNShimmer.m
//  shimmerTest
//
//  Created by Jose Padilla on 3/19/16.
//  Copyright © 2016 Facebook. All rights reserved.
//

#import "RNShimmer.h"
#import "RCTViewManager.h"

@interface RCT_EXTERN_MODULE(RNShimmerSwift, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(enabled, BOOL);
RCT_EXPORT_VIEW_PROPERTY(beginFadeDuration, double);
RCT_EXPORT_VIEW_PROPERTY(beginTime, double);
RCT_EXPORT_VIEW_PROPERTY(endFadeDuration, double);
RCT_EXPORT_VIEW_PROPERTY(pauseDuration, double);
RCT_EXPORT_VIEW_PROPERTY(animationOpacity, CGFloat);
RCT_EXPORT_VIEW_PROPERTY(highlightLength, CGFloat);
RCT_EXPORT_VIEW_PROPERTY(opacity, CGFloat);
RCT_EXPORT_VIEW_PROPERTY(speed, CGFloat);
RCT_EXPORT_VIEW_PROPERTY(direction, NSString);

@end

File: RNShimmer/RNShimmer.swift
//
//  RNShimmer.swift
//  shimmerTest
//
//  Created by Jose Padilla on 3/19/16.
//  Copyright © 2016 Facebook. All rights reserved.
//

import Foundation


@objc(RNShimmer)
class RNShimmer : FBShimmeringView {
 
  override init(frame: CGRect) {
    super.init(frame: frame);
    self.frame = frame;
  }
  
  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented");
  }
  
  override func layoutSubviews() {
    super.layoutSubviews();
    
    self.contentView = self.subviews[0];
  }
  
  func setEnabled(enabled: Bool) {
    if enabled == false {
      self.shimmering = false;
    } else {
      self.shimmering = true;
    }
  }
  
  func setAnimationOpacity(opacity: CGFloat) {
    self.shimmeringAnimationOpacity = opacity;
  }
  
  func setBeginFadeDuration(beginFadeDuration: Double) {
    self.shimmeringBeginFadeDuration = beginFadeDuration;
  }
  
  func setBeginTime(beginTime: Double) {
    self.shimmeringBeginTime = beginTime;
  }
  
  func setEndFadeDuration(endFadeDuration: Double) {
    self.shimmeringEndFadeDuration = endFadeDuration;
  }
  
  func setHighlightLength(highlightLength: CGFloat) {
    self.shimmeringHighlightLength = highlightLength;
  }
  
  func setSpeed(speed: CGFloat) {
    self.shimmeringSpeed = speed;
  }
  
  func setOpacity(opacity: CGFloat) {
    self.shimmeringOpacity = opacity;
  }
  
  func setPauseDuration(pauseDuration: Double) {
    self.shimmeringPauseDuration = pauseDuration;
  }
  
  func setDirection(direction: String) {
    switch(direction) {
      case "Up":
        self.shimmeringDirection = .Up;
        break;
      case "Right":
        self.shimmeringDirection = .Right;
        break;
      case "Down":
        self.shimmeringDirection = .Down;
        break;
      case "Left":
        self.shimmeringDirection = .Right;
        break;
      default:
        self.shimmeringDirection = .Right;
        break;
    }
  }
  
}